cdptools/indexers/indexer.py

```diff
@@ -96,6 +97,179 @@ def get_raw_transcript(transcript_path: Union[str, Path]) -> str:
+        max_span_size: int
+            The max contextual span size.
+            Default: 10
+
+        Returns
+        -------
+        span: str
+            The contextual sentence span surrounding the index provided.
+
+        Example
+        -------
+        ```
+        get_context_span_for_index(
+            terms=[
+                "Hello",
+                "and",
+                "good",
+                "morning."
+            ],
+            index=0
+        )
+
+        >> "Hello and good morning."
+        ```
+
+        ```
+        get_context_span_for_index(
+            terms=[
+                "But",
+                "I",
+                "don't",
+                "think",
+                "the",
+                "barrier",
+                "is",
+                "so",
+                "high",
+                "to",
+                "allow",
+                "individuals",
+                "to",
+                "use",
+                "the",
+                "service."
+            ],
+            index=0
+        )
+
+        >> "But I don't think the barrier is so high to ..."
+        ```
+
+        ```
+        get_context_span_for_index(
+            terms=[
+                "But",
+                "I",
+                "don't",
+                "think",
+                "the",
+                "barrier",
+                "is",
+                "so",
+                "high",
+                "to",
+                "allow",
+                "individuals",
+                "to",
+                "use",
+                "the",
+                "service."
+            ],
+            index=6
+        )
+
+        >> "... I don't think the barrier is so high to allow ..."
+        ```
+
+        """
+        # Check index provided
+        if index < 0:
+            raise IndexError("For simplicity's sake, we don't support negative index context span retrieval. Sorry.")
+
+        # Window to store valid terms in
+        window = deque()
+        window.append(terms[index])
+
+        # Check if the current term is the last in the sentence already
+        if Indexer.term_is_end_of_sentence(terms[index]):
+            right_side_valid = False
+        else:
+            right_side_valid = True
+            right_side_index = index + 1
+
+        # We can always check left side initially
+        left_side_valid = True
+        left_side_index = index - 1
+        current_exploration_direction = "left"
+
+        # Keep appending items to the window until it his max_span_size and both sides are still valid
+        while len(window) < max_span_size and (left_side_valid or right_side_valid):
+            # If the current exploration direction is leftwards and the left side is still valid, check the left term
+            if current_exploration_direction == "left" and left_side_valid:
+                # If the term is the end of the prior sentence, mark the current position as no longer valid to explore
```

Comment:
Could terms ever contain more than one sentence?
e.g. terms=["four", "words." ,"yes", "punctuation"] index=2
would return "yes punctuation ..." ?

Reply:
Yep that works! And that is generally how it should be used. Will add a test for it.